### DISCUSSION TITLE
/v2/portfolios/ returns empty array if no filter is provided

### DISCUSSION THREAD

**Discussion by clegendre** [amzn/ads-advanced-tools-docs#91] [Q&A]
Hello,

It looks like [portfolios endpoint](https://advertising.amazon.com/API/docs/en-us/reference/2/portfolios#/Portfolios/listPortfolios) always return an empty array if we don't provide the optional parameter `portfolioIdFilter`.
Is it working as expected?
How do we get the portfolioIds if we have to specify them to obtain the portfolios values?

**Comment by elizafarley** [comment]
Hi @clegendre, I was able to get portfolios to return without a filter, using a call like: 

```
curl --location 'https://advertising-api.amazon.com/v2/portfolios' \
--header 'Amazon-Advertising-API-ClientId: amzn1.application-oa2-client.xxxxxxxxxxxx' \
--header 'Authorization: Bearer xxxxxxxxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxxxxxx' \
```

Are you sure the profile you are using has active portfolios? If it does, you may be experiencing a bug, and I'd suggest [opening a JIRA ticket for support](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2).

**Reply by clegendre** [reply]
Thank you,
Yes I'm sure, while executing this one:
```
curl --location 'https://advertising-api.amazon.com/v2/portfolios' \
--header 'Amazon-Advertising-API-ClientId: amzn1.application-oa2-client.xxxxxxxxxxxxxxxxxxxxxxxxxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxxxxxxxxxxx' \
--header 'Authorization: Bearer xxxxxxxxxxxxx'
```

Result is : []

When I execute the following, : 
```
curl --location 'https://advertising-api.amazon.com/v2/portfolios/?portfolioIdFilter=1234%2C5678' \
--header 'Amazon-Advertising-API-ClientId: amzn1.application-oa2-client.xxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxx' \
--header 'Authorization: Bearer xxxx'
```
Result is
```json
[
    {
        "portfolioId": 1234,
        "name": "cccccc",
        "inBudget": true,
        "state": "enabled"
    },
    {
        "portfolioId": 5678,
        "name": "zzzzz",
        "inBudget": true,
        "state": "enabled"
    }
]
```

I'll open a ticket.
Thank you
